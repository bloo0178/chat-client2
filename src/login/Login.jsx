import React from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { withStyles } from "@material-ui/core/styles";

const styles = {
  loginContainer: {
    height: "100vh",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  }
};

class Login extends React.Component {
  state = {
    username: ""
  };

  handleChange = event => {
    this.setState({ username: event.target.value });
  };

  handleClick = async () => {
    const { history } = this.props;
    const { username } = this.state;
    if (!username) return;
    this.setState({ username: "" });
    //await login(this.state.username);
    history.push("/channels");
  };

  render() {
    const {
      classes: { loginContainer }
    } = this.props;
    return (
      <div className={loginContainer}>
        <h1>react.chat</h1>
        <TextField
          id="username"
          placeholder="Enter a username"
          margin="normal"
          label="Username"
          InputLabelProps={{ shrink: true }}
          value={this.state.username}
          onChange={this.handleChange}
        />
        <Button
          id="submit-username"
          variant="contained"
          color="primary"
          onClick={this.handleClick}
        >
          Submit
        </Button>
      </div>
    );
  }
}

export default withStyles(styles)(Login);
