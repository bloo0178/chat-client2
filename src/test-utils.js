import React from "react";
import { render } from "react-testing-library";
import { GlobalStateProvider } from "./common/GlobalState.context";

// https://github.com/kentcdodds/testing-workshop/blob/92c6d4d4de8ee1c351aca8e53010428ad1dd9a0c/client/test/til-client-test-utils.js#L8-L18

const customRender = (node, options) => {
  return render(<GlobalStateProvider>{node}</GlobalStateProvider>, options);
};

// re-export everything
export * from "react-testing-library";

// override render method
export { customRender as render };









/* PREVIOUS WORKING IMPELEMENTATION 

import React from "react";
import { render } from "react-testing-library";
import { GlobalStateProvider } from "./common/GlobalState.context";

const customRender = (node, options) => {
  return render(<GlobalStateProvider>{node}</GlobalStateProvider>, options);
};

// re-export everything
export * from "react-testing-library";

// override render method
export { customRender as render };

*/
