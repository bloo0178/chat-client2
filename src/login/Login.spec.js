import React from "react";
//import { mount, shallow } from "enzyme";
import { createShallow, createMount } from "@material-ui/core/test-utils";
import Login from "./Login";

describe("<Login />", () => {
  let shallow;
  let wrapper;
  //console.log(wrapper.find('div').debug()); <--need to place within function

  beforeAll(() => {
    shallow = createShallow({ dive: true });
    wrapper = shallow(<Login />);
  });

  it("should render correctly", () => expect(wrapper).toMatchSnapshot());

  it("should render a <div>", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should render an <h1>", () => {
    expect(wrapper.find("h1").length).toEqual(1);
  });

  describe("Username input", () => {
    it("should respond to change event and change the state of the Login Component", () => {
      wrapper
        .find("#username")
        .simulate("change", { target: { name: "username", value: "test" } });
      expect(wrapper.state("username")).toEqual("test");
    });
    it("should clear input upon submitting a valid non-null name", () => {
        wrapper
        .find('#username')
        .simulate("change", { target: { name: "username", value: "test" }});
        wrapper
        .find("#submit-username").simulate('click');
        expect(wrapper.state("username")).toEqual('');
    })
  });


});

describe("<Login /> (mounted)", () => {
  let wrapper;
  let mount;

  beforeAll(() => {
    mount = createMount();
    wrapper = mount(<Login />);
  });

  afterAll(() => {
    mount.cleanUp();
  });

  it("should work", () => {
    wrapper = mount(<Login />);
  });

 
});
