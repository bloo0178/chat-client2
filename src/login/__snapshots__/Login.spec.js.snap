// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    classes={
      Object {
        "loginContainer": "Login-loginContainer-1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          react.chat
        </h1>,
        <TextField
          InputLabelProps={
            Object {
              "shrink": true,
            }
          }
          id="username"
          label="Username"
          margin="normal"
          onChange={[Function]}
          placeholder="Enter a username"
          required={false}
          select={false}
          value=""
          variant="standard"
        />,
        <WithStyles(Button)
          color="primary"
          id="submit-username"
          onClick={[Function]}
          variant="contained"
        >
          Submit
        </WithStyles(Button)>,
      ],
      "className": "Login-loginContainer-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "react.chat",
        },
        "ref": null,
        "rendered": "react.chat",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "InputLabelProps": Object {
            "shrink": true,
          },
          "id": "username",
          "label": "Username",
          "margin": "normal",
          "onChange": [Function],
          "placeholder": "Enter a username",
          "required": false,
          "select": false,
          "value": "",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Submit",
          "color": "primary",
          "id": "submit-username",
          "onClick": [Function],
          "variant": "contained",
        },
        "ref": null,
        "rendered": "Submit",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            react.chat
          </h1>,
          <TextField
            InputLabelProps={
              Object {
                "shrink": true,
              }
            }
            id="username"
            label="Username"
            margin="normal"
            onChange={[Function]}
            placeholder="Enter a username"
            required={false}
            select={false}
            value=""
            variant="standard"
          />,
          <WithStyles(Button)
            color="primary"
            id="submit-username"
            onClick={[Function]}
            variant="contained"
          >
            Submit
          </WithStyles(Button)>,
        ],
        "className": "Login-loginContainer-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "react.chat",
          },
          "ref": null,
          "rendered": "react.chat",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "InputLabelProps": Object {
              "shrink": true,
            },
            "id": "username",
            "label": "Username",
            "margin": "normal",
            "onChange": [Function],
            "placeholder": "Enter a username",
            "required": false,
            "select": false,
            "value": "",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Submit",
            "color": "primary",
            "id": "submit-username",
            "onClick": [Function],
            "variant": "contained",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
}
`;
